### PR TITLE
Implement Self-CheckIn (EXPOSUREAPP-5747)

### PR DESCRIPTION
**Pull request by jurajkusnier**
The self-check-in on the organizer flow's event card implementation.

Attendee onboarding is possible to change in test menu > Onboarding Options > Event Creation Onboarding

### CODE REVIEW

**Review by harambasicluka — DISMISSED** [review]
LGTM, tested on a Xiaomi Mi 9 (Android 9) 🚀

**Comment by axelherbstreith** [comment]
When starting the app for the first time, adding a new event and then press "Self-Checkin" I get directly redirected to the screen that shows the event. The initial onboarding screen is jumped over and not shown, it only appears when opening "Check-Ins" the next time.

https://user-images.githubusercontent.com/75120552/113882246-6be95900-97bd-11eb-9acc-18b760b96c74.mp4

**Comment by jurajkusnier** [comment]
> When starting the app for the first time, adding a new event and then press "Self-Checkin" I get directly redirected to the screen that shows the event. The initial onboarding screen is jumped over and not shown, it only appears when opening "Check-Ins" the next time.

Good catch. I reused the existing deep-link navigation. So it should be the same as when the user scans QR-code. We should clarify if it's correct or not

**Comment by mtwalli** [comment]
> > When starting the app for the first time, adding a new event and then press "Self-Checkin" I get directly redirected to the screen that shows the event. The initial onboarding screen is jumped over and not shown, it only appears when opening "Check-Ins" the next time.
> 
> Good catch. I reused the existing deep-link navigation. So it should be the same as when the user scans QR-code. We should clarify if it's correct or not

I already spoke to @chris-cwa  about that, it should be the same as in ContactDairy shortcuts

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt [hunk 1]
```diff
@@ -43,10 +45,21 @@ class DeltaOnboardingFragmentViewModel @AssistedInject constructor(
 
     fun isDeltaOnboardingDone() = settings.wasInteroperabilityShownAtLeastOnce
 
-    fun setDeltaOboardinDone(value: Boolean) {
+    fun setDeltaOboardingDone(value: Boolean) {
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt:48]: typo `Obording` -> `Onboarding`

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt [hunk 2]
```diff
@@ -43,10 +45,21 @@ class DeltaOnboardingFragmentViewModel @AssistedInject constructor(
...
         settings.wasInteroperabilityShownAtLeastOnce = value
     }
 
+    fun isAttendeeOboardingDone() =
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt:52]: typo

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt [hunk 3]
```diff
@@ -43,10 +45,21 @@ class DeltaOnboardingFragmentViewModel @AssistedInject constructor(
...
+        traceLocationSettings.onboardingStatus == TraceLocationSettings.OnboardingStatus.ONBOARDED_2_0
+
+    fun setAttendeeOboardingDone(value: Boolean) {
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt:55]: typo

**Review by LukasLechnerDev — DISMISSED** [review]
LGTM!

Tested: 
- Self Check-in for newly created trace location
- Onboarding was shown before check-in
- While checked-in, it is not possible to check in again
- After check-out, it is possible to check-in again
- For subsequent check-ins, no Onboarding is shown

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt (hunk 1 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt]: fixed

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt (hunk 3 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt]: fixed

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt (hunk 2 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/deltaonboarding/ui/DeltaOnboardingFragmentViewModel.kt]: fixed

**Review by LukasLechnerDev — APPROVED** [review]

**Review by harambasicluka — APPROVED** [review]

**Pull request merged**